### ISSUE TITLE
Paket fails to resolve some .NET Standard packages

### ISSUE THREAD

**Issue by yaakov-h** [fsprojects/Paket#2287]
### Description

When using Paket with a Code Analyzer project (as per https://github.com/dotnet/roslyn/issues/18414#issuecomment-292028741), The Paket-modified NuGet restore target fails with an error.

### Repro steps

1. Create a .NET Standard library with the following project:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard1.3</TargetFramework>
    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
    <OutputPath>..\Bin\</OutputPath>
    <PackageTargetFallback>portable-net45+win8+wp8+wpa81;netcore45</PackageTargetFallback>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="2.0.0" />
  </ItemGroup>

  <Import Project="..\.paket\Paket.Restore.targets" />

</Project>
```

2. Add the following `paket.dependencies` file:

```
source https://api.nuget.org/v3/index.json
#cache \\some-cache-server\paket-cache

nuget NuGet.CommandLine
nuget NUnit
nuget stylecop.analyzers
nuget Microsoft.CodeAnalysis.Analyzers

nuget Microsoft.CodeAnalysis 2.0
nuget Microsoft.CodeAnalysis.Common 2.0
nuget Microsoft.CodeAnalysis.CSharp 2.0
nuget Microsoft.CodeAnalysis.CSharp.Workspaces 2.0
nuget Microsoft.CodeAnalysis.Workspaces.Common 2.0

nuget Microsoft.Composition 1.0.27

nuget System.Reflection.Metadata 1.4.2
nuget System.Collections.Immutable 1.3.1
nuget System.IO.FileSystem 4.3
```

3. Run `paket update`

4. Run `dotnet restore`

### Expected behavior

`dotnet restore` completes successfully.

### Actual behavior

`dotnet restore` fails with errors:

```
error : Package System.Security.Cryptography.OpenSsl 4.3.0 is not compatible with netstandard1.3 (.NETStandard,Version=v1.3). Package System.Security.Cryptography.OpenSsl 4.3.0 supports: netstandard1.6 (.NETStandard,Version=v1.6)
error : Package Microsoft.CodeAnalysis.Elfie 0.10.6 is not compatible with netstandard1.3 (.NETStandard,Version=v1.3). Package Microsoft.CodeAnalysis.Elfie 0.10.6 supports: net46 (.NETFramework,Version=v4.6)
error : One or more packages are incompatible with .NETStandard,Version=v1.3.
```

### Known workarounds

The projects builds if I remove `Paket.targets` from the csproj file.

**Comment by forki** [comment]
are you sure this is a paket issue? This is nuget speaking.

**Comment by yaakov-h** [comment]
It reliably works without Paket.targets imported and reliably fails with Paket.targets imported.

I don't see how it would be anything else.

**Comment by 0x53A** [comment]
The error message basically says it all: the packages only provide netstandard1.6, but you try to target netstandard1.3.

Paket does not support PackageTargetFallback https://github.com/NuGet/Home/wiki/PackageTargetFallback-(new-design-for-Imports), which provides a hack to override this.

**Comment by matthid** [comment]
@forki yes it is probably a problem how we delegate packages to nuget

**Comment by yaakov-h** [comment]
So can we make Paket support PackagesTargetFallback, or do I have to dump Paket for a Roslyn Analyzer project?

**Comment by forki** [comment]
> It reliably works without Paket.targets imported and reliably fails with Paket.targets imported.

not really. if you remove the targets file then there will be no dependencies included at all.

**Comment by yaakov-h** [comment]
@forki when removing the targets file it appears to fall back to the standard .NET SDK method of handling dependencies with the `<PackageReference>` element in the project.

**Comment by forki** [comment]
yes but when you want to use paket then you don't have the PackageReference elements. you would need to delete them and reference via paket.references

**Comment by yaakov-h** [comment]
OK so I've switched it up by removing the `PackageReference`s and creating `paket.references` which contains nothing but `Microsoft.CodeAnalysis.CSharp.Workspaces`.

Now using `Paket version 5.0.0-alpha020`.

`paket.dependencies` looks like this:

```
source https://api.nuget.org/v3/index.json

nuget NuGet.CommandLine
nuget NUnit
nuget StyleCop.Analyzers
nuget Microsoft.CodeAnalysis.Analyzers

nuget Microsoft.CodeAnalysis 2.0
nuget Microsoft.CodeAnalysis.Common 2.0
nuget Microsoft.CodeAnalysis.CSharp 2.0
nuget Microsoft.CodeAnalysis.CSharp.Workspaces 2.0
nuget Microsoft.CodeAnalysis.Workspaces.Common 2.0
```

Same build errors when running `dotnet restore`::

```
      Package Microsoft.CodeAnalysis.Elfie 0.10.6 is not compatible with netstandard1.3 (.NETStandard,Version=v1.3). Package Microsoft.CodeAnalysis.Elfie 0.10.6 supports: net46 (.NETFramework,Version=v4.6)
      Package System.Security.Cryptography.OpenSsl 4.3.0 is not compatible with netstandard1.3 (.NETStandard,Version=v1.3). Package System.Security.Cryptography.OpenSsl 4.3.0 supports: netstandard1.6 (.NETStandard,Version=v1.6)
      One or more packages are incompatible with .NETStandard,Version=v1.3.
```

It looks like I can work around it by putting the version fallback into `paket.references` like so:

```
Microsoft.CodeAnalysis.CSharp.Workspaces
Microsoft.Composition framework: portable-net45+win8+wp8+wpa81
```

**Comment by yaakov-h** [comment]
Closing as resolved - this works with Paket 5.1.6.
